File: clients/kmp/shared/src/androidMain/kotlin/org/wyrdsekai/app/engine/persistence/InMemoryVitalityStore.kt
package org.wyrdsekai.app.engine.persistence

import org.wyrdsekai.app.engine.agent.VitalityState

/** In-memory vitality store stub for Android (Phase 2b). */
class InMemoryVitalityStore : VitalityStore {
    private val store = mutableMapOf<String, VitalityState>()
    override suspend fun save(entityId: String, state: VitalityState) { store[entityId] = state }
    override suspend fun load(entityId: String): VitalityState? = store[entityId]
}

File: clients/rn/src/engine/persistence/InMemoryVitalityStore.ts
/**
 * In-memory VitalityStore — no platform deps, useful for tests and mobile default.
 */

import type { VitalityState } from '../agent/VitalityState';
import type { VitalityStore } from './VitalityStore';

export class InMemoryVitalityStore implements VitalityStore {
  private store = new Map<string, VitalityState>();

  async save(entityId: string, state: VitalityState): Promise<void> {
    this.store.set(entityId, { ...state });
  }

  async load(entityId: string): Promise<VitalityState | null> {
    const s = this.store.get(entityId);
    return s ? { ...s } : null;
  }

  clear(): void {
    this.store.clear();
  }
}
